### DISCUSSION TITLE
Flex grid questions

### DISCUSSION THREAD

**Discussion by viniciusccarvalho** [quasarframework/quasar#15021] [General - Components / Directives / etc] [answered]
Hi folks. I always struggle with flex grid positioning :( I find myself many times just using manual (non responsive spacing) to fix stuff. 

For instance on this example: https://codepen.io/viniciusccarvalho/pen/QWxVRav?editors=101

Can someone help me out understand:

1. On a flex row, does elements inside `col` children always take the full width? In my example I would like to have the q-select to be left aligned but not take the full width. Is my only option control the `max-width` style?
2. For the element on the second column (a q-pagination) I tried several classes to make it float-right or align at the end, with no success, what is the correct way of moving it to the end of the div? 

Thanks and apologies for the super dumb question.

**Comment by pdanpdan** [comment, marked as answer]
https://codepen.io/pdanpdan/pen/MWXPvyg?editors=1010

**Reply by viniciusccarvalho** [reply]
Thank you!

**Reply by viniciusccarvalho** [reply]
Once again, thank you very much for the above and beyond example, with multiple choices :) 

May I ask a few follow up questions?

1. What is the `col-grow` class? I couldn't find a reference on the search page of quasar. Curious about what it does.
2. I noticed in one of the examples a div with both col and row classes were used. I'm not sure I understand why, when would such need arise?

Thanks

**Comment by pdanpdan** [comment]
`.flex`, `.column` and `.row` are setting how the element is displayed as a container for it's children
`.col...` classes are setting how the element is displayed as a children of a container

so `.column.col` means display element as a column container for it's children and size it as a col in its parent

`.col-grow` means it will try to stretch the element as much as it can (`.col` tries to make it equal to other sibling `.col`s)
